content/docs/develop/userscripts/about-userscripts.md

```diff
@@ -0,0 +1,142 @@
+---
+title: About userscripts
+description: Userscripts are JavaScript files that are executed every time the user loads a Scratch page. They can modify the document's HTML, add new buttons, customize Scratch editor behavior, and so much more.
+aliases: 
+  - /docs/developing/about-userscripts
+---
+
+Userscripts are JavaScript files that are executed every time the user loads a Scratch page. They can modify the document's HTML, add new buttons, customize Scratch editor behavior, and so much more.
+
+Similarly to userscripts that you might download for userscript managers like Tampermonkey or Greasemonkey, Scratch Addons userscripts consist of pieces of JavaScript that are executed in the same execution context as the JavaScript code from Scratch itself. In browser extension vocabulary, this execution context is often named "main world".
+
+Even though Scratch Addons userscripts are part of a browser extension, they cannot access any `chrome.*` or `browser.*` APIs. Instead, Scratch Addons offers an [`addon.*` API](/docs/reference/addon-api/). 
+
+
+## Declaring userscripts in the addon manifest
+
+{{< admonition warning >}}
+**Some changes require an extension reload** from `chrome://extensions` to take effect, such as updating the addon manifest file.
+
+It's not necessary to reload the extension when changing the source of an already existing userscript JavaScript file. In those cases, reloading the page is enough.
+{{< /admonition >}}
+
+<!-- TODO: revamp -->
+
+<!-- TODO: create a "matches" page -->
+
+Go to the manifest of your addon (addon.json) and add a property called `userscripts"`.  
+This property must be an array.  
+Each item of the array must have the following properties: `"url"` and `"matches"`.  
+`"url"` must be a relative URL to a JavaScript file.  
+`"matches"` must be an array of URLs where you want to run the userscript on. You can use asterisks.
```

Comment:
[matches](https://scratchaddons.com/docs/reference/addon-manifest/#matches) doesn't have to be a URL. Also, regular expressions are supported, right?

Reply:
This is part of the old doc page, and I haven't updated that part yet. Unfortunately, I had to rename the file, so Git can't tell the difference and it looks like the whole article is new.

Reply:
Updated the "matches" documentation in the "addon manifest" page.